### PR TITLE
Cleanup before doc merge

### PR DESCRIPTION
**Pull request by nneveu**
# Summary of Changes: 

- Moved older documentation to sphinx source location
- Cleaned up formatting and imports 
- Added Sphinx to requirements.txt 
- Removed constants in data analysis directory 
- Removed uneeded directories in matlab2py dir

# Notes
These small changes were made in preparation for the bigger documentation changes in PR #120

### CODE REVIEW

**Comment by nneveu** [comment]
Of course I forgot to run the tests. Fixing those now.

**Comment by nneveu** [comment]
Test are now cleaned up too.

File: docs/source/cor_plot.rst
Comment (eloiseyang) [inline docs/source/cor_plot.rst:1]: Our current naming convention for Correlation Plot in Matlab is to shorten the title as `corrPlot`. It would be nice to maintain that convention by spelling `cor_plot` as `corr_plot`, or even `corrPlot` if you're okay with camel case. I know it would be a drag to track down every `cor_plot` in the code, so please feel free to skip this change.

**Review by eloiseyang — APPROVED** [review]
I appreciate how you simplified the file structure here. Thank you for getting a start on translating Matlab to Python, I think this will be a really good tool for the HLA team. I just left one minor comment on the naming convention `cor_plot` vs `corr_plot`, but I feel these changes could be merged as is.

File: tests/unit_tests/lcls_tools/common/matlab2py/image/__init__.py
Comment (phys-cgarnier) [inline tests/unit_tests/lcls_tools/common/matlab2py/image/__init__.py:1]: Is this file needed to do unittests?

**Review by phys-cgarnier — APPROVED** [review]

**Review by phys-cgarnier — COMMENTED** [review]
Hi, I just wanted to verify that those deleted files are no longer needed, everything else looks good.

File: docs/source/cor_plot.rst
Comment (nneveu) [reply docs/source/cor_plot.rst:1]: I think corr_plot is reasonable, happy to get that in this PR. Thanks for the catch!

File: tests/unit_tests/lcls_tools/common/matlab2py/image/__init__.py
Comment (nneveu) [reply tests/unit_tests/lcls_tools/common/matlab2py/image/__init__.py:1]: __init__ files are needed in every folder for imports. It's part of the package set up.

File: tests/unit_tests/lcls_tools/common/data_analysis/fitting/test_fit_gaussian.py
[the remark this answers is not in the document]
Comment (nneveu) [reply tests/unit_tests/lcls_tools/common/data_analysis/fitting/test_fit_gaussian.py:1]: I want to deprecate it. This test file mostly has bare bones test set up/template and two bucket tests. I'm not sure these fit into the new structure/image code that was written. This is a good comment though, I'm going to go back and double check the original file to see if there are any functions we should keep.

File: docs/source/cor_plot.rst
Comment (nneveu) [reply docs/source/cor_plot.rst:1]: name updated, and tests passed.

File: tests/unit_tests/lcls_tools/common/data_analysis/fitting/test_fit_gaussian.py
[the remark this answers is not in the document]
Comment (nneveu) [reply tests/unit_tests/lcls_tools/common/data_analysis/fitting/test_fit_gaussian.py:1]: Yea after a second look, I still want to remove these. Some of the functions are in your new fitting tool, some can be replaced with functions in scipy (peak finding), the rest are specific to some image processing tricks. We can always look at this on old hashes/commit if we decide we want a piece of it.
